extension/user/auth-strategy/facebook.js

```diff
@@ -12,6 +13,40 @@ module.exports = async (context, input) => {
     return {}
   }
 
-  context.log.error('Facebook strategy is not implemented')
-  throw new InvalidCredentialsError()
+  const clientCredentials = context.config.credentials
+  const authUrl = context.config.magentoUrl + '/auth/token'
+  const { parameters: { success, profile } } = input
+  const log = context.log
+  const storages = context.storage
+  const request = context.tracedRequest('magento-user-extension:facebook-login', { log: true })
+
+  if (!success) {
+    throw new InvalidCredentialsError()
+  }
+
+  const th = new TokenHandler(clientCredentials, authUrl, storages, log, request, !context.config.allowSelfSignedCertificate)
+
+  const magentoTokenResponse = await login(th, profile, input.strategy)
+  await new Promise((resolve, reject) => {
+    th.deleteGuestTokens((err) => {
+      if (err) return reject(err)
+      resolve()
+    })
+  })
+
+  return { userId: profile.email, magentoTokenResponse }
+
+  /**
+   * @param {TokenHandler} tokenHandler
+   * @param {UserLoginInputParameters} userCredentials
+   * @param {string} strategy
+   */
+  async function login (tokenHandler, userCredentials, strategy) {
+    return new Promise((resolve, reject) => {
```

Comment:
same as above, as long as the CB's are not preventing us. Although it would be nice to refactor a little :D